include/shark/Algorithms/DirectSearch/Operators/ReferenceVectorAdaptation.h

```diff
@@ -0,0 +1,58 @@
+#ifndef SHARK_ALGORITHMS_DIRECT_SEARCH_OPERATORS_REFERENCE_VECTOR_ADAPTATION_H
+#define SHARK_ALGORITHMS_DIRECT_SEARCH_OPERATORS_REFERENCE_VECTOR_ADAPTATION_H
+
+#include <shark/Algorithms/DirectSearch/Operators/Selection/ReferenceVectorGuidedSelection.h>
+
+namespace shark {
+
+struct ReferenceVectorAdaptation
+{
+	typedef shark::Individual<RealVector, RealVector> IndividualType;
+
+	void operator()(
+		std::vector<IndividualType> const & population,
+		RealMatrix & referenceVectors,
+		RealVector & minAngles)
+	{
+		typedef ReferenceVectorGuidedSelection rv;
+		const RealMatrix f = rv::extractPopulationFitness(population);
```

Comment:
we already have a nice helper function which does something similar. The only difference is that it returns a range of vectors and no matrix:

auto f = penalizedFitness(population);

implementation is in Individual.h